### PR TITLE
fix: File name of svg downloaded from disco preserves context including sample name and gene

### LINKED ISSUE #319: When downloading disco plot, improve SVG file name
raised by https://jira.opensciencedatacloud.org/browse/SV-2250, quote `suggest we label the downloaded SVG as same as plot title "MMRF_1596_1_BM_CD138pos, with TP53 V216M.svg"`

since this kind of file name is about some context when disco plot is launched, i can simply pass in a `svgFileName` attribute in disco parameter for disco code to use

### PR DESCRIPTION
**Pull request by xzhou82**
## Description

closes #319 
test at http://localhost:3000/?genome=hg38&gene=ENST00000407796&mds3=GDC, click a mutation and launch disco. first the "prioritize by CGC" is checked by default. click Download, the svg file name carries sample name and gene etc

launch disco from non-gdc (pnet) works

mds3 manual tests pass

all 781 CI tests pass

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/plots/disco/Disco.ts [hunk 1]
```diff
@@ -15,23 +15,28 @@ import { RingType } from './ring/RingType.ts'
...
+	private type: string
+	private opts: any
 	private state: any
 	private id: any
 	private app: any
 
 	constructor(opts: any) {
 		this.type = 'Disco'
 		this.opts = opts
-		this.discoInteractions = new DiscoInteractions(this)
```
Comment (xzhou82) [inline client/plots/disco/Disco.ts:29]: @aacic this line is moved to main(), as DiscoInteractions requires this.state. please verify this change is okay

**Review by karishma-gangwani — APPROVED** [review]
so the filename carries it. but do we also show the sample, and gene name on the svg?

File: client/plots/disco/Disco.ts (hunk 1 above)
Comment (aacic) [reply client/plots/disco/Disco.ts:29]: It's fine. It's not super expensive to create the DiscoInteractions object in main.

File: client/plots/matrix.interactivity.js [hunk 2]
```diff
@@ -186,7 +186,12 @@ export function setInteractivity(self) {
 						self.state.vocab.dslabel,
 						sample,
 						sandbox.body,
-						self.app.opts.genome
+						self.app.opts.genome,
+						{
+							label: {
+								prioritizeGeneLabelsByGeneSets: true // TODO control this at dataset-level
```
Comment (aacic) [inline client/plots/matrix.interactivity.js:192]: Is this a leftover? Or do you want to merge it?

**Comment by xzhou82** [comment]
not yet. since Alex is okay i will merge. later can show sample/gene on
plot directly

**Pull request merged**